Installing-and-Configuring/OS/Generic.md

```diff
@@ -0,0 +1,258 @@
+ hooks, you also need keyscript to be in `/etc/crypttab` even as a no-op
+ `/bin/cat`:
+
+`sda5_crypt UUID=XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX /secret.key luks,keyscript=/bin/cat`
+
+(Credit to [https://www.pavelkogan.com/2015/01/25/linux-mint-encryption/](https://www.pavelkogan.com/2015/01/25/linux-mint-encryption/)
+ for this trick).
+
+bootloader
+---
+
+heads reads /boot/grub.cfg in order to boot your system.  Since heads is acting as the bootloader you do not need to install grub to your MBR.  It won't hurt if you do since the MBR is mostly ignored.  If you choose to not install grub make sure you generate the grub.cfg file or heads cannot boot your system.
+
+
+Default Boot
+====
+
+Heads will prompt you to set a default boot after OS installation.
+
+Boot config files
+===
+
+A user has the option to make persistent modifications to the non-Qubes boot
+ process by creating one or more of the following files:
+
+| file | description |
+| ---- | ---- |
+| kexec_menu.txt | contains multiple options for parameters to the kexec command|
+| kexec_hashes.txt | a sha256sum file from within the respective boot directory |
+| kexec_iso_add.txt | a sh variable to override the standard ISO kernel argument
+ additions |
+| kexec_iso_remove.txt | a sh variable to override the standard ISO kernel
+ argument removals |
+| kexec_default.$N.txt | specifies the default kexec parameters corresponding to
+ the Nth menu option |
+| kexec_default_hashes.txt | a sha256sum file for the default entry kexec file
+ parameters |
+| kexec_rollback.txt | a sha256sum of the TPM counter contents in the tmp
+ directory |
+| kexec_key_devices.txt | contains a list of "device uuid" combos for all LUKS
+ devices to unlock |
+| kexec_key_lvm.txt | contains the name of an LVM group to activate on boot |
+
+These can be placed in any of the following locations:
+
+| location | description |
+| ---- | ---- |
+| /boot/ | used during internal HD boot |
+| /media/ | used during standard USB boot |
+| /media/kexec_iso/$ISO_FILENAME/ | used during USB boot from a particular ISO
+ file |
+
+These files are only used if there is an appropriate signature for them in
+ `kexec.sig` covering all `kexec*.txt` in that location.  This can be generated
+  by running `kexec-sign-config -p /boot/`, etc.  These files are copied by
+ `kexec-check-config` to `/tmp/kexec/` only there's a valid signature.  From
+  there the boot routines reference only the configs in `/tmp/kexec`.
+
+
+Dynamic vs Persistent Boot Options
+====
+
+There are two ways for heads to boot Operating systems from /boot.
+
+* Dynamic (no kexec_menu.txt)
+* Persistent
+
+If there is no persistent `kexec_menu.txt`, the boot directory will be searched
+ for grub/syslinux-like configurations and it will be generated dynamicly (for
+ any of the HD/USB/USB-ISO locations).  Creating a persistent
+ `kexec_menu.txt` can be useful to limit the options displayed or to make
+ persistent alterations to xen or kernel params.
```

Comment:
A user guide should inform the reader of her options and the implications of choosing one of those options.  In this example, the user may manually create kexec_menu.txt or let heads use the dynamic method.  

We could move the docs to a separate page and reference them here noting that the default is the dynamic option.  The challenge is then 'where to put that new page?'  It would need a new section at the same level as OS , Recovery, upgrading.  Perhaps "Configuring Boot Options" under "Operating Systems"

Reply:
WIP  I separated the boot options and linked in generic.